arpeggio/__init__.py

```diff
@@ -1017,15 +1017,18 @@ class Terminal(ParseTreeNode):
             name in the case of an error node.
         suppress(bool): If True this terminal can be ignored in semantic
             analysis.
+        extra_value(object): additional information (e.g. the re matcher
+            object)
     """
 
     __slots__ = ['rule', 'rule_name', 'position', 'error', 'comments',
                  'value', 'suppress']
```

Comment:
`extra_info` attribute should be added to `__slots__` for [Python optimization purposes](https://stackoverflow.com/questions/472000/usage-of-slots).

Reply:
Ah. I am just reading about these slots....
Adapted.

Reply:
Note: Somehow these slots seems to be not "active" (no error occurred when I stored the extra_info into - probably - __dict__). Probably because the base class does not define __slots__...

Reply:
Yes, I think you are right. This should be checked. I think Python will silently revert to "normal" way of creating objects (using `__dict__`) if something is set incorrectly.

Reply:
Since parse trees can grow quite large using `__slots__` can lead to a significant optimization. But it is unrelated to this PR.